apps/studio/src/components/sidebar/ConnectionSidebar.vue

```diff
@@ -217,29 +268,43 @@ const log = rawLog.scope('connection-sidebar');
         return _.orderBy(this.connectionConfigs, this.sortOrder)
       },
       components() {
-        return [
-          this.$refs.savedConnectionList,
-          this.$refs.recentConnectionList
-        ]
+        if (!this.noPins) {
+          return [
+            this.$refs.pinnedConnectionList,
+            this.$refs.savedConnectionList,
+            this.$refs.recentConnectionList
+          ]
+        } else {
+          return [
+            this.$refs.savedConnectionList,
+            this.$refs.recentConnectionList
+          ]
+        }
       }
     },
     mounted() {
-      this.split = Split(this.components, {
-        elementStyle: (dim, size) => ({
-          'flex-basis': `calc(${size}%)`
-        }),
-        direction: 'vertical',
-        sizes: this.sizes
-      })
+      this.split = this.getSplit()
     },
     methods: {
+      getSplit() {
+        return Split(this.components, {
+          elementStyle: (dim, size) => ({
+            'flex-basis': `calc(${size}%)`
+          }),
+          direction: 'vertical',
+          sizes: this.noPins ? [50, 50] : [33, 33, 33]
+        })
+      },
       importFromLocal() {
         console.log("triggering import")
         this.$root.$emit(AppEvent.promptConnectionImport)
       },
       refresh() {
         this.$store.dispatch('data/connectionFolders/load')
         this.$store.dispatch('data/connections/load')
+        this.$store.dispatch('pinnedConnections/loadPins').then(() => this.$store.dispatch('pinnedConnections/reorder', this.pinnedConnections))
+        this.split.destroy()
+        this.split = this.getSplit()
```

Comment:
You don't need to call it explicitly, it will be called with the watcher.

In Vue, it's best to rely on watchers to dispatch events whenever possible, that way it's harder to forget something if you make changes in several different places.